apps/io-func-wallet-solution-backend/src/infra/azure/cosmos/nonce.ts

```diff
@@ -19,4 +19,19 @@ export class CosmosDbNonceRepository implements NonceRepository {
       (error) => new Error(`Error inserting nonce: ${error}`)
     );
   }
+
+  delete(nonce: string) {
+    return TE.tryCatch(
+      async () => {
+        await this.#container.item(nonce, nonce).delete();
+      },
+      (error) =>
+        typeof error === "object" &&
```

Comment:
what about a zod interface instead?

Reply:
Should we include zod just for this parsing?

Reply:
nope, I see you're using io-ts here, you may want to consider it

Reply:
I think it would become more complicated. This is a 'non fp-ts' part of code

Reply:
why?